1 Niagara 4 Overview Exercise - BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,240 @@
+package com.tridiumuniversity.myFile;
+
+import javax.baja.nre.annotations.NiagaraAction;
+import javax.baja.nre.annotations.NiagaraProperty;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.status.BStatus;
+import javax.baja.status.BStatusNumeric;
+import javax.baja.status.BStatusString;
+import javax.baja.sys.Action;
+import javax.baja.sys.BAbsTime;
+import javax.baja.sys.BComponent;
+import javax.baja.sys.BMonth;
+import javax.baja.sys.Flags;
+import javax.baja.sys.Property;
+import javax.baja.sys.Slot;
+import javax.baja.sys.Sys;
+import javax.baja.sys.Type;
+
+@NiagaraType
+@NiagaraProperty(
+        name = "fileName",
+        type = "BStatusString",
+        defaultValue = "new BStatusString (\"\", BStatus.nullStatus)",
+        flags = Flags.SUMMARY
+)
+
+@NiagaraProperty(
+        name = "path",
+        type = "String",
+        defaultValue = "\"path\"",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "size",
+        type = "BStatusNumeric",
+        defaultValue = "new BStatusNumeric(0, BStatus.nullStatus)",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "createdDate",
+        type = "BAbsTime",
+        defaultValue = "BAbsTime.make(2023, BMonth.make(5), 28, 12, 34, 14, 3)",
```

Comment:
The problem with using a specific date for the default value of a `BAbsTime` is that it means that every copy of your `BFile` class is going to have that date as the default value for its `createdDate` property. If that's what you want, then great! But a lot of times that's not what you had in mind, so it's better for the `defaultValue` to be something like `BAbsTime.DEFAULT`.